pkg/occlient/occlient.go

```diff
@@ -1116,23 +1119,44 @@ func (c *Client) WaitAndGetPod(selector string) (*corev1.Pod, error) {
 		return nil, errors.Wrapf(err, "unable to watch pod")
 	}
 	defer w.Stop()
-	for {
-		val, ok := <-w.ResultChan()
-		if !ok {
-			break
-		}
-		if e, ok := val.Object.(*corev1.Pod); ok {
-			glog.V(4).Infof("Status of %s pod is %s", e.Name, e.Status.Phase)
-			switch e.Status.Phase {
-			case corev1.PodRunning:
-				glog.V(4).Infof("Pod %s is running.", e.Name)
-				return e, nil
-			case corev1.PodFailed, corev1.PodUnknown:
-				return nil, errors.Errorf("pod %s status %s", e.Name, e.Status.Phase)
+
+	podChannel := make(chan *corev1.Pod)
+	watchErrorChannel := make(chan error)
+
+	go func() {
+	loop:
+		for {
+			val, ok := <-w.ResultChan()
```

Comment:
I was thinking if the podchannel and watchErrorChannel can be avoided if we move the logic of listening to the w.ResultChan to the select below?

But it seems w.ResultChan will get updates for every update to pod's attributes... and that would cause select to complete :(

Reply:
Yes exactly.